test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
+      ### VERIFY START ###
+      within('#dialog') do
+        # error messages in dialog
+        assert_text I18n.t('projects.samples.transfers.create.error')
+        # colon is removed from translation in UI
+        assert_text I18n.t('services.samples.transfer.sample_exists', sample_puid: @sample30.puid,
+                                                                      sample_name: @sample30.name).gsub(':', '')
       end
-    end
 
-    test 'user with maintainer access should be able to see the transfer samples button' do
-      user = users(:joan_doe)
-      login_as user
+      # verify sample1 and 2 transferred, sample 30 did not
+      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary.one', count: 1, locale: @user.locale))
+      assert_no_selector "tr[id='#{@sample1.id}']"
+      assert_no_selector "tr[id='#{@sample2.id}']"
+      assert_selector "tr[id='#{@sample30.id}']"
 
-      visit namespace_project_samples_url(@namespace, @project)
-
-      assert_selector 'a', text: I18n.t('projects.samples.index.transfer_button'), count: 1
-    end
-
-    test 'user with guest access should not be able to see the transfer samples button' do
-      user = users(:ryan_doe)
-      login_as user
-
-      visit namespace_project_samples_url(@namespace, @project)
-
-      assert_selector 'a', text: I18n.t('projects.samples.index.transfer_button'), count: 0
-    end
-
-    test 'user should not be able to see the edit button for the sample' do
-      user = users(:ryan_doe)
-      login_as user
-
-      visit namespace_project_sample_url(@namespace, @project, @sample1)
-
-      assert_selector 'a', text: I18n.t('projects.samples.show.edit_button'), count: 0
-    end
-
-    test 'user should not be able to see the remove button for the sample' do
-      user = users(:ryan_doe)
-      login_as user
-
-      visit namespace_project_sample_url(@namespace, @project, @sample1)
-
-      assert_selector 'a', text: I18n.t('projects.samples.index.remove_button'), count: 0
+      # destination project
+      visit namespace_project_samples_url(namespace, project25)
+      assert_text strip_tags(I18n.t(:'viral.pagy.limit_component.summary', from: 1, to: 4, count: 4,
+                                                                           locale: @user.locale))
+      assert_selector "tr[id='#{@sample1.id}']"
+      assert_selector "tr[id='#{@sample2.id}']"
+      assert_no_selector "tr[id='#{@sample30.id}']"
+      ### VERIFY END ###
     end
 
-    test 'visiting the index should  not allow the current user only edit action' do
-      user = users(:joan_doe)
-      login_as user
+    test 'sample transfer project listing should be empty for maintainer if no other projects in hierarchy' do
+      ### SETUP START ###
+      login_as users(:user28)
+      namespace = groups(:group_hotel)
+      project = projects(:projectHotel)
+      visit namespace_project_samples_url(namespace, project)
```

Comment:
**nitpick (blocking):** Can you assert for limit component before proceeding to reduce potential for flakes.

Reply:
Added in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac